ios/src/CountlyReactNative.m

```diff
@@ -812,4 +886,77 @@ + (void) log: (NSString *) theMessage{
     resolve(@"Remote Config Cleared.");
   });
 }
+
+RCT_EXPORT_METHOD(apm:(NSArray*)arguments) {
+  dispatch_async(dispatch_get_main_queue(), ^ {
+    config.enablePerformanceMonitoring = YES;
+  });
+}
+RCT_EXPORT_METHOD(startTrace:(NSArray*)arguments) {
+    dispatch_async(dispatch_get_main_queue(), ^ {
+        NSString* traceKey = [arguments objectAtIndex:0];
+        [Countly.sharedInstance startCustomTrace: traceKey];
+    });
+}
+RCT_EXPORT_METHOD(cancelTrace:(NSArray*)arguments) {
+    dispatch_async(dispatch_get_main_queue(), ^ {
+        NSString* traceKey = [arguments objectAtIndex:0];
+        [Countly.sharedInstance cancelCustomTrace: traceKey];
+    });
+}
+RCT_EXPORT_METHOD(clearAllTrace:(NSArray*)arguments) {
+    dispatch_async(dispatch_get_main_queue(), ^ {
+        [Countly.sharedInstance clearAllCustomTraces];
+    });
+}
+RCT_EXPORT_METHOD(endTrace:(NSArray*)arguments) {
+    dispatch_async(dispatch_get_main_queue(), ^ {
+        NSString* traceKey = [arguments objectAtIndex:0];
+        NSMutableDictionary *metrics = [[NSMutableDictionary alloc] init];
+        for(int i=1,il=(int)arguments.count;i<il;i+=2){
+            metrics[[arguments objectAtIndex:i]] = [arguments objectAtIndex:i+1];
+        }
+        [Countly.sharedInstance endCustomTrace: traceKey metrics: metrics];
+    });
+}
+RCT_EXPORT_METHOD(startNetworkRequest:(NSArray*)arguments) {
+    // NSString* networkTraceKey = [arguments objectAtIndex:0];
+    NSString* uniqueId = [arguments objectAtIndex:1];
+    int startTime = [self getTime];
+    if(networkRequest == nil){
+        networkRequest = [[NSMutableDictionary alloc] init];
+    }
+    [networkRequest setValue: [NSNumber numberWithInt: startTime] forKey:uniqueId];
+}
+RCT_EXPORT_METHOD(endNetworkRequest:(NSArray*)arguments) {
```

Comment:
This should reworked to just have "recordNetworkTrace". A call for android will be added, for now a placeholder should be used.